### ISSUE TITLE
"redirectUrl" field in /nativecheckout/v2/transactions response

### ISSUE THREAD

**Issue by anzap** [VivaPayments/API#2109]
Hi All,
We encountered today some issues with viva integration on our end. The response of `/nativecheckout/v2/transactions` call contains a field `redirectUrl`, which was not there before. In our case this was returned with value `null`.
Unfortunately i cannot seem to find any information in documentation about what this field is and if we need to do anything about it.
Can you share some feedback on this?

Thanks.

**Comment by matthewrgourd** [comment]
Hi @anzap, are you working in the demo environment or on live? In the demo environment, I just tried a POST to `https://demo-api.vivapayments.com/nativecheckout/v2/chargetokens` with test card number 5188340000000011 and the field was successfully populated (see below).

REQUEST
```
POST /nativecheckout/v2/chargetokens HTTP/1.1
Host: demo-api.vivapayments.com
Content-Type: application/json
Authorization: Bearer eyJhbGciOiJSUzI1NiIsImtpZCI6IkRZLTQxdTNzOWNsLTlTZkFZV3V5TXpxMlg1OCIsInR5cCI6IkpXVCIsIng1dCI6IkRZLTQxdTNzOWNsLTlTZkFZV3V5TXpxMlg1OCJ9.eyJuYmYiOjE2MTEzODQyMjcsImV4cCI6MTYxMTM4NzgyNywiaXNzIjoiaHR0cHM6Ly9kZW1vLWFjY291bnRzLnZpdmFwYXltZW50cy5jb20iLCJhdWQiOlsiaHR0cHM6Ly9kZW1vLWFjY291bnRzLnZpdmFwYXltZW50cy5jb20vcmVzb3VyY2VzIiwiY29yZV9hcGkiXSwiY2xpZW50X2lkIjoiOWZ3OXV4bmxnY2FhczFqYXlkdzM0ZGxiODVvNDY2OW44OTRnMXpiNDgyOGE3LmFwcHMudml2YXBheW1lbnRzLmNvbSIsInVybjp2aXZhOnBheW1lbnRzOmNsaWVudF9wZXJzb25faWQiOiI0QTE0MEMyOC00MEEyLTQ2MzMtOEM3NC01MUU3MTJDNzA0QTUiLCJzY29wZSI6WyJ1cm46dml2YTpwYXltZW50czpjb3JlOmFwaTphY3F1aXJpbmciLCJ1cm46dml2YTpwYXltZW50czpjb3JlOmFwaTphY3F1aXJpbmc6Y2FyZHM6dG9rZW5zIiwidXJuOnZpdmE6cGF5bWVudHM6Y29yZTphcGk6bmF0aXZlY2hlY2tvdXR2MiJdfQ.KJniGS2rstIDDUirxgfDBDlWLdsx3eQjIIh_CpwfTVu6LO12pEFwGx27x2pDp5G7AySgqagHj5VTTDjW0j9Tvi5YUS4UgYLK0qXHhpRHYVaHvQefzn3HPxd87NxCNQnRqdRh6FSzSDheV9wW9sX6TpmsMAzS7SekROBZSbPAaaK5BWC1e9VMb27T21C8ReyPqGQO_UBOO5KzDBQ42kFho6lGoC2zaMEtEkkl8NVIYHtgcR6bwuN8Gqab_bkaPejx1BX5ICjHwE3GGtFN8IanoOoHl0nawln1eE6T8X6-r0ecziMryQgPNGXozGPbHSwFpdLvQrUMF3GOibT4LLSxlQ
Content-Length: 211

{
    "Amount": 100,
    "Cvc": "111",
    "Number": "5188340000000011",
    "Holdername": "John Doe",
    "ExpirationYear": 2030,
    "ExpirationMonth": 10,
    "SessionRedirectUrl": "https://www.example.com"
}
```
RESPONSE
```
{
    "chargeToken": "ctok_AJ7_usfxvoYPC5MSg2DYt8JT278",
    "redirectToACSForm": "<iframe id=\"tdsMmethodTgtFrame\" name=\"tdsMmethodTgtFrame\" style=\"width: 1px; height: 1px; visibility: hidden;\" src=\"javascript:false;\" xmlns=\"http://www.w3.org/1999/xhtml\">\n<!--.-->\n</iframe><form id=\"tdsMmethodForm\" name=\"tdsMmethodForm\" action=\"https://3ds-acs.test.modirum.com/mdpayacs/3ds-method\" method=\"post\"  xmlns=\"http://www.w3.org/1999/xhtml\">\n<input type=\"hidden\" name=\"3DSMethodData\" value=\"eyAidGhyZWVEU1NlcnZlclRyYW5zSUQiIDogImYwMzc1NWMwLTNkNTAtNTkxMy04MDAwLTAwMmVhNmQwMWUyOSIsICJ0aHJlZURTTWV0aG9kTm90aWZpY2F0aW9uVVJMIiA6ICJodHRwczovL2RlbW8udml2YXBheW1lbnRzLmNvbS93ZWIvY2hlY2tvdXQvdGhyZWVkc2VjdXJlYXV0aGVudGljYXRpb24vODQyZjVkNjctNDJjZS00MjQ2LTk4NGEtMzQ0Njg3MWRkZDNmIiB9\"/>\n<input type=\"hidden\" name=\"threeDSMethodData\" value=\"eyAidGhyZWVEU1NlcnZlclRyYW5zSUQiIDogImYwMzc1NWMwLTNkNTAtNTkxMy04MDAwLTAwMmVhNmQwMWUyOSIsICJ0aHJlZURTTWV0aG9kTm90aWZpY2F0aW9uVVJMIiA6ICJodHRwczovL2RlbW8udml2YXBheW1lbnRzLmNvbS93ZWIvY2hlY2tvdXQvdGhyZWVkc2VjdXJlYXV0aGVudGljYXRpb24vODQyZjVkNjctNDJjZS00MjQ2LTk4NGEtMzQ0Njg3MWRkZDNmIiB9\"/>\n</form><script type=\"text/javascript\" xmlns=\"http://www.w3.org/1999/xhtml\">\ndocument.getElementById(\"tdsMmethodForm\").submit();\n</script>"
}
```

**Comment by rigasp** [comment]
Hello @matthewrgourd 
- This is about Production
- This happened yesterday around 16.10 Eastern European time
- I see @anzap is refering to redirectURL not sessionRedirectURL
- The question is also that this field is not part of VivaWallet documentation so it seems the response has changed. 
- Besides the fact that this is null, it caused an issue as our application was not expecting it

**Comment by matthewrgourd** [comment]
Hi @rigasp, thanks for jumping in with your clarification. May I ask if you're working with @anzap? Can I have a full example of a response that shows the `redirectURL` parameter?

**Comment by matthewrgourd** [comment]
Hi @anzap, nothing has changed at this end as far as I know. Can you let me know the request body for the call to `/nativecheckout/v2/transactions` that returned the `redirectUrl` null value? I'm struggling to understand this as I can't reproduce it.

**Comment by anzap** [comment]
Hey @matthewrgourd, sorry for the late response.
This is the response we got from viva while posting to `/nativecheckout/v2/transactions` endpoint

```
{"transactionId":"b499612b-a7d1-4bde-bfc1-XXXX","redirectUrl":null}"
```
`redirectUrl` was the problem for us, but we updated our code to ignore any unknown fields and since then we did not have an issue. Just wondering if this is something we should take care of in a different way.

@rigasp is indeed working with me on the project.
